src/clients/UBAClient/UBAClientUtilities.ts

```diff
@@ -235,23 +301,21 @@ export async function updateUBAClient(
       },
     };
 
+    // Grab all bundle ranges for this chain. This logic is isolated into a function that we can unit test.
+    const bundleBounds = getMostRecentBundleBlockRanges(
+      chainId,
+      maxBundleStates,
+      hubPoolBlockNumber,
+      hubPoolClient,
+      spokePoolClients
+    );
+
+    // Now, for each token, load the bundle data including:
+    // - valid flows in the bundle
+    // - opening running balance
+    // - opening incentive balance
+    // - UBA config to apply when charging fees to flows in bundle
     const tokenStates = await relevantTokenSymbols.reduce(async (accumulator, tokenSymbol) => {
-      // Find the last MAX_BUNDLE_LOOKBACK_SIZE bundle start/end blocks for this token
-      let startingBlock = hubPoolBlockNumber;
-      const bundleBounds: { start: number; end: number }[] = [];
-      for (let i = 0; i < maxBundleStates; i++) {
-        const lastPreviousBlock = hubPoolClient.getLatestBundleEndBlockForChain(
-          relevantChainIds,
-          startingBlock,
-          chainId
-        );
-        // Push the structure to the start of the list
-        bundleBounds.unshift({
-          start: startingBlock,
-          end: lastPreviousBlock,
-        });
-        startingBlock = lastPreviousBlock - 1;
-      }
```

Comment:
Nit: this logic trims the bundle blocks so that we only have N of them in memory. Should we refactor the code to include it in a different location

Reply:
What do you propose? I thought the goal was to have only `maxBundleStates` bundles total?

I'm also curious why you use `.unshift` instead of `.push`?